### ISSUE TITLE
Can't create items in collections with join field, draft/autosave enabled, and using Postgres adapter

### ISSUE THREAD

**Issue by mikeploythai** [payloadcms/payload#9365]
### Describe the Bug

When creating a new item in a collection with a join field and autosave enabled, it redirects to the `/collections/{collectionName}/create` route as expected, but then redirects immediately to the new item's page (`/collections/{collectionName}/{itemId}`), as if the item were already created. Since the item doesn't exist at this point, it shows the "Not Found" UI.

If draft is allowed, but not autosave, the creation UI works as expected. This error also does not happen, even with autosave enabled, if I use a different db adapter, such as SQLite.

### Link to the code that reproduces this issue

https://github.com/mikeploythai/payload-bug

### Reproduction Steps

1. Visit the [Vercel deployment](https://payload-bug.vercel.app/admin) of the repo (auto-login enabled)
2. Create a new item in the `services` collection
3. Be presented with the "Not Found" screen

https://github.com/user-attachments/assets/f2b054b0-d0fa-401a-9c29-aaf7a9705a9d

### Which area(s) are affected? (Select all that apply)

db-postgres

### Environment Info

```text
Binaries:
  Node: 22.11.0
  npm: 10.9.0
  Yarn: N/A
  pnpm: 9.12.3
Relevant Packages:
  payload: 3.0.1
  next: 15.0.0
  @payloadcms/db-postgres: 3.0.1
  @payloadcms/email-nodemailer: 3.0.1
  @payloadcms/graphql: 3.0.1
  @payloadcms/next/utilities: 3.0.1
  @payloadcms/payload-cloud: 3.0.1
  @payloadcms/richtext-lexical: 3.0.1
  @payloadcms/translations: 3.0.1
  @payloadcms/ui/shared: 3.0.1
  react: 19.0.0-rc-65a56d0e-20241020
  react-dom: 19.0.0-rc-65a56d0e-20241020
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.1.0: Thu Oct 10 21:02:26 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T8122
  Available memory (MB): 16384
  Available CPU cores: 8
```

**Comment by mikeploythai** [comment]
> If draft is allowed, but not autosave, the creation UI works as expected.

An update to this, the creation UI loads, but once I save the draft, the actual content does not get saved. I then get redirected to the new item's page (`/collections/{collectionName}/{itemId}`), but get shown the "Not Found" error once again.

**Comment by jacobsfletch** [comment]
@mikeploythai this is likely fixed in #9365

**Comment by mikeploythai** [comment]
Awesome! Thanks 🫡
